File: python/121-Best-Time-To-Buy-and-Sell-Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res = 0
        
        l = 0
        for r in range(1, len(prices)):
            if prices[r] < prices[l]:
                l = r
            res = max(res, prices[r] - prices[l])
        return res

File: swift/121-Best-Time-to-Buy-and-Sell-Stock.swift
class Solution {
    func maxProfit(_ prices: [Int]) -> Int {
        var bestPrice = 0
        guard !prices.isEmpty else { return bestPrice }
        var curDay = prices[0]
        for i in 0..<prices.count {
            let price = prices[i]
            if curDay > price {
                curDay = price
            } else {
                if (price - curDay) > bestPrice {
                    bestPrice = price - curDay
                }
            }
        }
        return bestPrice
    }
}
